spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainerTests.java

```diff
@@ -564,6 +567,24 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
 		container.stop();
 	}
 
+	@Test
+	public void testContainerNotRecoveredAfterExhaustingRecoveryBackOff() throws Exception {
+		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(mock(ConnectionFactory.class));
+		container.setQueueNames("foo");
+		container.setRecoveryBackOff(new FixedBackOff(100, 3));
+		container.setConcurrentConsumers(10);
+		container.afterPropertiesSet();
+		container.start();
```

Comment:
Yeah... It was there with `Mocks` from the beginning, but after that I found that it works as is with such a bad config for `ConnectionFactory` :smile: 
Will fix